cmd/cluster/powervs/create.go

```diff
@@ -133,18 +134,19 @@ func applyPlatformSpecificsValues(ctx context.Context, exampleOptions *apifixtur
 			infraID = infraid.New(opts.Name)
 		}
 		opt := &powervsinfra.CreateInfraOptions{
-			BaseDomain:             opts.BaseDomain,
-			ResourceGroup:          opts.PowerVSPlatform.ResourceGroup,
-			InfraID:                infraID,
-			OutputFile:             opts.InfrastructureJSON,
-			PowerVSRegion:          opts.PowerVSPlatform.Region,
-			PowerVSZone:            opts.PowerVSPlatform.Zone,
-			PowerVSCloudInstanceID: opts.PowerVSPlatform.CloudInstanceID,
-			PowerVSCloudConnection: opts.PowerVSPlatform.CloudConnection,
-			VpcRegion:              opts.PowerVSPlatform.VpcRegion,
-			Vpc:                    opts.PowerVSPlatform.Vpc,
+			BaseDomain:      opts.BaseDomain,
+			ResourceGroup:   opts.PowerVSPlatform.ResourceGroup,
+			InfraID:         infraID,
+			OutputFile:      opts.InfrastructureJSON,
+			Region:          opts.PowerVSPlatform.Region,
+			Zone:            opts.PowerVSPlatform.Zone,
+			CloudInstanceID: opts.PowerVSPlatform.CloudInstanceID,
+			CloudConnection: opts.PowerVSPlatform.CloudConnection,
+			VPCRegion:       opts.PowerVSPlatform.VPCRegion,
+			VPC:             opts.PowerVSPlatform.VPC,
+			Debug:           opts.PowerVSPlatform.Debug,
 		}
-		infra = &powervsinfra.Infra{}
+		infra = &powervsinfra.Infra{ID: opts.InfraID}
```

Comment:
why are we feeding this ID here and what change it will bring this now with this change?

Reply:
This is required if we create infra with `create cluster` command `infra.ID` is not set, because in `create infra` we are setting `infra.ID` https://github.com/openshift/hypershift/blob/main/cmd/infra/powervs/create.go#L211 in `Run()`. Setting this here since we are calling `SetupInfra` directly.